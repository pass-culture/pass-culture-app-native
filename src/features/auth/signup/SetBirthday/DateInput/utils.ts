export interface DateValidation {
  isComplete: boolean
  isValid: boolean
  isDateAboveMin: boolean
  isDateBelowMax: boolean
}

export interface DateInputRef {
  clearFocuses: () => void
}
