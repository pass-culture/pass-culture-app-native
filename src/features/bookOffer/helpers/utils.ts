import { OfferStockResponse } from 'api/gen'

export enum OfferStatus {
  BOOKABLE = 'BOOKABLE',
  NOT_BOOKABLE = 'NOT_BOOKABLE',
  NOT_OFFERED = 'NOT_OFFERED',
}

export const formatToKeyDate = (dateString: Date | string) => {
  const date = new Date(dateString)
  const day = ('0' + date.getDate()).slice(-2)
  const month = ('0' + (date.getMonth() + 1)).slice(-2)
  const year = date.getFullYear()
  return `${year}-${month}-${day}`
}

export const formatHour = (dateString: string | null | undefined): string => {
  if (!dateString) return ''
  const date = new Date(dateString)
  const hours = date.getHours()
  const minutes = ('0' + date.getMinutes()).slice(-2)
  return `${hours}:${minutes}`
}

export const getStatusFromStocks = (stocks: OfferStockResponse[], credit: number): OfferStatus => {
  const bookableStocks = stocks.filter(({ isBookable }) => isBookable)
  if (bookableStocks.length === 0) return OfferStatus.NOT_BOOKABLE
  return bookableStocks.some(({ price }) => price <= credit)
    ? OfferStatus.BOOKABLE
    : OfferStatus.NOT_BOOKABLE
}
