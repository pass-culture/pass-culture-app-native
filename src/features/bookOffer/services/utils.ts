import { OfferStockResponse } from 'api/gen'
import { formatToSlashedFrenchDate } from 'libs/dates'
import { CENTS_IN_EURO } from 'libs/parsers/pricesConversion'

const EURO_SYMBOL = '€'

export const getStocksByDate = (
  stocks: OfferStockResponse[]
): { [date: string]: OfferStockResponse[] } => {
  const stockDates: { [date: string]: OfferStockResponse[] } = {}

  stocks.forEach((stock) => {
    if (stock.beginningDatetime !== null && stock.beginningDatetime !== undefined) {
      const formattedDate = formatToSlashedFrenchDate(stock.beginningDatetime.toString())
      stockDates[formattedDate] = stockDates[formattedDate]
        ? stockDates[formattedDate].concat(stock)
        : [stock]
    }
  })

  return stockDates
}

export enum OfferStatus {
  BOOKABLE = 'BOOKABLE',
  NOT_BOOKABLE = 'NOT_BOOKABLE',
  NOT_OFFERED = 'NOT_OFFERED',
}

export const formatToFrenchDecimalWithoutSpace = (cents: number) => {
  const euros = cents / CENTS_IN_EURO
  // we show 2 decimals if price is not round. Ex: 21,50€
  const fixed = euros === Math.floor(euros) ? euros : euros.toFixed(2)
  return `${fixed.toString().replace('.', ',')}${EURO_SYMBOL}`
}

export const getDateStatusAndPrice = (
  date: Date,
  stocksDates: { [date: string]: OfferStockResponse[] }
): { status: OfferStatus; price: string | null } => {
  const stocksByDate = stocksDates[formatToSlashedFrenchDate(date.toString())]
  if (!stocksByDate) return { status: OfferStatus.NOT_OFFERED, price: null }

  const prices = stocksByDate.map((stock) => stock.price)
  const price = formatToFrenchDecimalWithoutSpace(Math.min(...prices))
  const offerIsBookable = stocksByDate.some((stock) => stock.isBookable)

  if (offerIsBookable) return { status: OfferStatus.BOOKABLE, price }
  return { status: OfferStatus.NOT_BOOKABLE, price }
}
