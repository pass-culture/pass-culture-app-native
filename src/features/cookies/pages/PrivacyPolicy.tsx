import React, { useEffect, useRef } from 'react'
import { ScrollView } from 'react-native'

import { ConsentState } from 'features/cookies/enums'
import { useCookies } from 'features/cookies/helpers/useCookies'
import { useIsCookiesListUpToDate } from 'features/cookies/helpers/useIsCookiesListUpToDate'
import { CookiesConsent } from 'features/cookies/pages/CookiesConsent'
import { AnchorProvider } from 'ui/components/anchor/AnchorContext'
import { useModal } from 'ui/components/modals/useModal'

export function PrivacyPolicy() {
  const { cookiesConsent: hasUserMadeCookieChoiceV2 } = useCookies()
  const {
    visible: cookiesConsentModalVisible,
    hideModal: hideCookiesConsentModal,
    showModal: showCookiesConsentModal,
  } = useModal(false)
  const { isCookiesListUpToDate } = useIsCookiesListUpToDate()
  const scrollViewRef = useRef<ScrollView>(null)
  const scrollYRef = useRef<number>(0)

  const handleCheckScrollY = useRef(() => {
    return scrollYRef.current
  }).current

  useEffect(() => {
    switch (hasUserMadeCookieChoiceV2.state) {
      case ConsentState.LOADING:
        break

      case ConsentState.UNKNOWN:
        showCookiesConsentModal()
        break

      default:
        if (isCookiesListUpToDate) {
          hideCookiesConsentModal()
        } else {
          showCookiesConsentModal()
        }
    }
  }, [
    hasUserMadeCookieChoiceV2,
    hideCookiesConsentModal,
    isCookiesListUpToDate,
    showCookiesConsentModal,
  ])

  return (
    <AnchorProvider scrollViewRef={scrollViewRef} handleCheckScrollY={handleCheckScrollY}>
      <CookiesConsent visible={cookiesConsentModalVisible} hideModal={hideCookiesConsentModal} />
    </AnchorProvider>
  )
}
