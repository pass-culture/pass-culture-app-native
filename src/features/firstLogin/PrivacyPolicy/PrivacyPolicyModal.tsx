import { t } from '@lingui/macro'
import { NavigationContainerRef } from '@react-navigation/native'
import React, { useCallback, useState, FunctionComponent, RefObject } from 'react'
import styled from 'styled-components/native'

import { openExternalUrl } from 'features/navigation/helpers'
import { env } from 'libs/environment'
import { _ } from 'libs/i18n'
import { ButtonPrimary } from 'ui/components/buttons/ButtonPrimary'
import { ButtonPrimaryWhite } from 'ui/components/buttons/ButtonPrimaryWhite'
import { ButtonTertiary } from 'ui/components/buttons/ButtonTertiary'
import { AppModal } from 'ui/components/modals/AppModal'
import { Close } from 'ui/svg/icons/Close'
import { ExternalLinkSite } from 'ui/svg/icons/ExternalLinkSite'
import { ColorsEnum, Spacer, Typo } from 'ui/theme'

export interface Props {
  visible: boolean
  onApproval: () => void
  onRefusal: () => void
  navigationRef?: RefObject<NavigationContainerRef>
  disableBackdropTap?: boolean
}

export const PrivacyPolicyModal: FunctionComponent<Props> = ({
  navigationRef,
  visible,
  onApproval,
  onRefusal,
  disableBackdropTap = true,
}) => {
  const [isVisible, setIsVisible] = useState<boolean>(visible)

  const goToConsentSettings = useCallback(() => {
    navigationRef?.current?.navigate('ConsentSettings', {
      onGoBack: () => {
        setIsVisible(true)
        if (navigationRef?.current?.canGoBack()) {
          navigationRef?.current?.goBack()
        }
      },
    })
    setIsVisible(false)
  }, [navigationRef])

  async function openCookiesPolicyExternalUrl() {
    await openExternalUrl(env.COOKIES_POLICY_LINK)
  }

  return (
    <AppModal
      visible={isVisible}
      title={_(t`Respect de ta vie privée`)}
      rightIcon={Close}
      onRightIconPress={onRefusal}
      disableBackdropTap={disableBackdropTap}>
      <Description>
        <Typo.Body>
          {_(
            t`Nous utilisons des outils pour réaliser des statistiques de navigation et offrir une experience plus sûre. En cliquant sur Continuer, tu acceptes l'utilisation de ces services détaillés dans notre`
          )}
        </Typo.Body>
      </Description>
      <ButtonTertiary
        title={_(t`Politique des cookies`)}
        onPress={openCookiesPolicyExternalUrl}
        icon={ExternalLinkSite}
        textSize={12}
      />
      <SubDescription>
        <Typo.Caption color={ColorsEnum.GREY_DARK}>
          {_(t`Tu peux modifier tes paramètres de confidentialité ici ou dans la page profil.`)}
        </Typo.Caption>
      </SubDescription>
      <ButtonPrimary title={_(t`Continuer`)} onPress={onApproval} />
      <Spacer.Column numberOfSpaces={2} />
      <ButtonPrimaryWhite
        title={_(t`Paramètres de confidentialité`)}
        onPress={goToConsentSettings}
      />
    </AppModal>
  )
}

const Description = styled.Text({
  textAlign: 'center',
})

const SubDescription = styled.Text({
  textAlign: 'center',
  paddingVertical: 20,
  paddingHorizontal: '10%',
})
