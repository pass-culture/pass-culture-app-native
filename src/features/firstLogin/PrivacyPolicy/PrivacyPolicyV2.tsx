import React, { useEffect } from 'react'

import { ConsentState } from 'features/cookies/enums'
import { useCookies } from 'features/cookies/helpers/useCookies'
import { useIsCookiesListUpToDate } from 'features/cookies/helpers/useIsCookiesListUpToDate'
import { CookiesConsent } from 'features/cookies/pages/CookiesConsent'
import { useModal } from 'ui/components/modals/useModal'

export function PrivacyPolicyV2() {
  const { cookiesConsent: hasUserMadeCookieChoiceV2 } = useCookies()
  const {
    visible: cookiesConsentModalVisible,
    hideModal: hideCookiesConsentModal,
    showModal: showCookiesConsentModal,
  } = useModal(false)
  const isCookiesListUpToDate = useIsCookiesListUpToDate()

  useEffect(() => {
    switch (hasUserMadeCookieChoiceV2.state) {
      case ConsentState.LOADING:
        break

      case ConsentState.UNKNOWN:
        showCookiesConsentModal()
        break

      default:
        if (isCookiesListUpToDate) {
          hideCookiesConsentModal()
        } else {
          showCookiesConsentModal()
        }
    }
  }, [
    hasUserMadeCookieChoiceV2,
    hideCookiesConsentModal,
    isCookiesListUpToDate,
    showCookiesConsentModal,
  ])

  return <CookiesConsent visible={cookiesConsentModalVisible} hideModal={hideCookiesConsentModal} />
}
