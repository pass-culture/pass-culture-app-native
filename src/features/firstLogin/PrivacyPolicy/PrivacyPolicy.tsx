import { NavigationContainerRef } from '@react-navigation/native'
import React, { RefObject, useEffect, useState } from 'react'

import { PrivacyPolicyModal } from 'features/firstLogin/PrivacyPolicy/PrivacyPolicyModal'
import { firebaseAnalytics } from 'libs/analytics'
import { storage } from 'libs/storage'

interface Props {
  navigationRef?: RefObject<NavigationContainerRef>
}

export function PrivacyPolicy(props: Props) {
  const [hasUserMadeCookieChoice, setHasUserMadeCookieChoice] = useState(true)

  useEffect(() => {
    storage.readObject('has_accepted_cookie').then((hasAcceptedCookie) => {
      if (hasAcceptedCookie === null) {
        setHasUserMadeCookieChoice(false)
      }
    })
  }, [])

  function acceptCookie() {
    setHasUserMadeCookieChoice(true)
    storage.saveObject('has_accepted_cookie', true)
  }

  function refuseCookie() {
    setHasUserMadeCookieChoice(true)
    storage.saveObject('has_accepted_cookie', false).then(() => {
      firebaseAnalytics.setAnalyticsCollectionEnabled(false)
    })
  }

  return hasUserMadeCookieChoice ? null : (
    <PrivacyPolicyModal
      visible={true}
      onApproval={acceptCookie}
      onRefusal={refuseCookie}
      navigationRef={props.navigationRef}
      disableBackdropTap
    />
  )
}
