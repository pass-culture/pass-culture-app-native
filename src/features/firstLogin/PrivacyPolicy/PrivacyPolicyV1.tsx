import React, { useEffect, useState } from 'react'

import { PrivacyPolicyModal } from 'features/firstLogin/PrivacyPolicy/PrivacyPolicyModal'
import { analytics } from 'libs/firebase/analytics'
import { getCookiesConsent, setCookiesConsent } from 'libs/trackingConsent/consent'
import { requestIDFATrackingConsent } from 'libs/trackingConsent/useTrackingConsent'

export function PrivacyPolicyV1() {
  const [hasUserMadeCookieChoiceV1, setHasUserMadeCookieChoiceV1] = useState(true)

  useEffect(() => {
    getCookiesConsent().then((hasAcceptedCookie) => {
      if (hasAcceptedCookie === null) {
        setHasUserMadeCookieChoiceV1(false)
      }
    })
  }, [])

  async function acceptCookie() {
    setHasUserMadeCookieChoiceV1(true)
    setCookiesConsent(true)
    await requestIDFATrackingConsent()
  }

  async function refuseCookie() {
    setHasUserMadeCookieChoiceV1(true)
    await setCookiesConsent(false)
    await analytics.logHasRefusedCookie()
    await analytics.disableCollection()
    await requestIDFATrackingConsent()
  }

  if (hasUserMadeCookieChoiceV1) return null

  return (
    <PrivacyPolicyModal
      visible
      onApproval={acceptCookie}
      onRefusal={refuseCookie}
      disableBackdropTap
    />
  )
}
