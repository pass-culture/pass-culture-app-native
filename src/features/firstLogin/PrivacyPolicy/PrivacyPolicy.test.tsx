import { act, fireEvent, render } from '@testing-library/react-native'
import React from 'react'

import { storage } from 'libs/storage'
import { flushAllPromises } from 'tests/utils'

import { PrivacyPolicy } from './PrivacyPolicy'

describe('<PrivacyPolicy />', () => {
  afterEach(() => {
    storage.clear('has_accepted_cookie')
  })

  it('should render privacy policy', async () => {
    jest.useFakeTimers()
    const renderAPI = render(<PrivacyPolicy />)
    await act(flushAllPromises)

    jest.advanceTimersByTime(1000)

    expect(renderAPI).toMatchSnapshot()
    jest.useRealTimers()
  })

  it('should show modal when has_accepted_cookie is null', async () => {
    const renderAPI = render(<PrivacyPolicy />)
    await act(flushAllPromises)

    const title = renderAPI.queryByText('Respect de ta vie privée')
    expect(title).toBeTruthy()
  })

  it('should NOT show modal when has_accepted_cookie is false', async () => {
    await storage.saveObject('has_accepted_cookie', false)
    const renderAPI = render(<PrivacyPolicy />)
    await act(flushAllPromises)

    const title = renderAPI.queryByText('Respect de ta vie privée')
    expect(title).toBeFalsy()
  })

  it('should NOT show modal when has_accepted_cookie is true', async () => {
    await storage.saveObject('has_accepted_cookie', true)
    const renderAPI = render(<PrivacyPolicy />)
    await act(flushAllPromises)

    const title = renderAPI.queryByText('Respect de ta vie privée')
    expect(title).toBeFalsy()
  })

  it('should close modal and set has_accepted_cookie to true in storage on dismiss', async () => {
    expect(await storage.readObject('has_accepted_cookie')).toBeNull()
    const renderAPI = render(<PrivacyPolicy />)
    await act(flushAllPromises)

    expect(await storage.readObject('has_accepted_cookie')).toBeNull()
    const title = renderAPI.queryByText('Respect de ta vie privée')
    expect(title).toBeTruthy()

    fireEvent.press(renderAPI.getByText('Continuer'))
    await act(flushAllPromises)

    expect(await storage.readObject('has_accepted_cookie')).toBe(true)
    expect(renderAPI.queryByText('Continuer')).toBeFalsy()
  })
})
