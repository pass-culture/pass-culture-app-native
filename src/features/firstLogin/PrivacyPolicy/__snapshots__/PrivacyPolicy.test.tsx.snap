// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivacyPolicy /> should render privacy policy 1`] = `
Array [
  <View
    opacity={1}
    style={
      Object {
        "backgroundColor": "#00000080",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100%",
        "justifyContent": "flex-end",
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": 49,
      }
    }
  />,
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    statusBarTranslucent={true}
    testID="modal"
    transparent={true}
    visible={true}
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "flexDirection": "column",
          "flexGrow": 1,
          "height": "100%",
          "justifyContent": "flex-end",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="click-away-area"
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderTopEndRadius": 16,
            "borderTopStartRadius": 16,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "maxHeight": "85%",
            "opacity": 1,
            "paddingBottom": 20,
            "paddingLeft": 20,
            "paddingRight": 20,
            "paddingTop": 20,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "width": "100%",
                "zIndex": 1,
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "flex-start",
                "flexBasis": 0,
                "flexGrow": 0.1,
                "flexShrink": 1,
                "marginLeft": -4,
                "opacity": 1,
              }
            }
            testID="leftIconButton"
          />
          <Text
            numberOfLines={2}
            style={
              Array [
                Object {
                  "color": "#151515",
                  "flexBasis": 0,
                  "flexGrow": 0.8,
                  "flexShrink": 1,
                  "fontFamily": "Montserrat-Medium",
                  "fontSize": 18,
                  "lineHeight": 22,
                  "textAlign": "center",
                },
              ]
            }
          >
            Respect de ta vie privée
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "flex-end",
                "flexBasis": 0,
                "flexGrow": 0.1,
                "flexShrink": 1,
                "marginRight": -4,
                "opacity": 1,
              }
            }
            testID="rightIconButton"
          >
            <View
              height={32}
              testID="rightIcon"
              width={32}
            >
              <Text>
                rightIcon-SVG-Mock
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "maxWidth": 500,
                "paddingTop": 20,
                "width": "100%",
              },
              Object {
                "paddingBottom": 8,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "textAlign": "center",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#151515",
                    "fontFamily": "Montserrat-Regular",
                    "fontSize": 15,
                    "lineHeight": 20,
                  },
                ]
              }
            >
              Nous utilisons des outils pour réaliser des statistiques de navigation et offrir une experience plus sûre. En cliquant sur Continuer, tu acceptes l'utilisation de ces services détaillés dans notre
            </Text>
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderRadius": 24,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
                "maxWidth": 500,
                "opacity": 1,
                "paddingBottom": 2,
                "paddingLeft": 2,
                "paddingRight": 2,
                "paddingTop": 2,
                "width": "100%",
              }
            }
            testID="button-container"
          >
            <View
              height={16}
              testID="button-icon"
              width={15}
            >
              <Text>
                button-icon-SVG-Mock
              </Text>
            </View>
            <Text
              adjustsFontSizeToFit={false}
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#eb0055",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 12,
                    "lineHeight": 20,
                    "marginLeft": 5,
                    "maxWidth": 650,
                  },
                ]
              }
              testID="button-title"
              textColor="#eb0055"
              textSize={12}
            >
              Politique des cookies
            </Text>
          </View>
          <Text
            style={
              Array [
                Object {
                  "paddingHorizontal": "10%",
                  "paddingVertical": 20,
                  "textAlign": "center",
                },
              ]
            }
          >
            <Text
              color="#626262"
              style={
                Array [
                  Object {
                    "color": "#626262",
                    "fontFamily": "Montserrat-SemiBold",
                    "fontSize": 12,
                    "lineHeight": 16,
                  },
                ]
              }
            >
              Tu peux modifier tes paramètres de confidentialité ici ou dans la page profil.
            </Text>
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#eb0055",
                "borderRadius": 24,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
                "maxWidth": 500,
                "opacity": 1,
                "paddingBottom": 2,
                "paddingLeft": 2,
                "paddingRight": 2,
                "paddingTop": 2,
                "width": "100%",
              }
            }
            testID="button-container"
          >
            <Text
              adjustsFontSizeToFit={false}
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 15,
                    "lineHeight": 20,
                    "marginLeft": 5,
                    "maxWidth": 650,
                  },
                ]
              }
              testID="button-title"
              textColor="#ffffff"
            >
              Continuer
            </Text>
          </View>
          <View
            numberOfSpaces={2}
            style={
              Array [
                Object {
                  "height": 8,
                },
              ]
            }
          />
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderRadius": 24,
                "borderWidth": 0,
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
                "maxWidth": 500,
                "opacity": 1,
                "paddingBottom": 2,
                "paddingLeft": 2,
                "paddingRight": 2,
                "paddingTop": 2,
                "width": "100%",
              }
            }
            testID="button-container"
          >
            <Text
              adjustsFontSizeToFit={false}
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#eb0055",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 15,
                    "lineHeight": 20,
                    "marginLeft": 5,
                    "maxWidth": 650,
                  },
                ]
              }
              testID="button-title"
              textColor="#eb0055"
            >
              Paramètres de confidentialité
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>,
]
`;
